tools/ci_build/github/apple/default_vision_os_framework_build_settings.json

```diff
@@ -0,0 +1,29 @@
+{
+    "build_osx_archs": {
+        "xros": [
+            "arm64"
+        ],
+        "xrsimulator": [
+            "arm64"
+        ]
+    },
+    "build_params": {
+        "base": [
+            "--parallel",
+            "--build_apple_framework",
+            "--use_xcode",
+            "--use_coreml",
+            "--skip_tests",
+            "--compile_no_warning_as_error",
```

Comment:
why do we need this? should we fix the warnings instead?

Reply:
If I recall correctly, this was added mainly to skip warnings like these (treated as error in the build:)

intermediates/xros_arm64/Debug/build/onnxruntime_common.build/Debug-xros/Objects-normal/arm64/cpuid_info.o
In file included from /Users/rachelguo/Repos/onnxruntime/onnxruntime/core/common/cpuid_info.cc:3:
/Users/rachelguo/Repos/onnxruntime/onnxruntime/core/common/cpuid_info.h:126:8: error: private field 'pytorch_cpuinfo_init_' is not used [-Werror,-Wunused-private-field]

already disabled cpu info in the cmake flags.

Reply:
can it be removed? it might hide warnings that we should address.

Reply:
oh, I see that this warning is coming from our code.
could place `[[maybe_unused]]` on that data member to suppress just that warning.
https://github.com/microsoft/onnxruntime/blob/a7787a0bad1291759be521050861cb374d6d9156/onnxruntime/core/common/cpuid_info.h#L126

@snnn do you think that would be fine?